docs/Plugins-Guide.md

```diff
@@ -271,6 +271,28 @@ module.exports = fp(dbPlugin)
 ```
 You can also tell to `fastify-plugin` to check the installed version of Fastify, in case of you need a specific api.
 
+As we mentioned earlier, Fastify starts loading its plugins __after__ `.listen()`, `.inject()` or `.ready()` are called and as such, __after__ they have been declared. This means that, even though the plugin may inject variables to the global scope via `decorate`, the decorated variables will not be accessible before calling `.listen()`, `.inject()` or `.ready()`.
+
+In case you rely on a variable injected by a preceding plugin and want to pass that in the `options` argument of `register`, you can do so by using a function instead of an object:
+```js
+const fastify = require('fastify')
```

Comment:
```suggestion
const fastify = require('fastify')()
```

Reply:
Integrated in the latest commit, I always make that mistake when creating snippets with `fastify`